storage/retrieval.go

```diff
@@ -135,3 +137,45 @@ func (storage *Bucket) GetTestLogLines(context context.Context, buildId string,
 	// Merge everything together
 	return NewMergingIterator(testChunkIterator, buildChunkIterator), nil
 }
+
+func (b *Bucket) FindBuildByID(ctx context.Context, id string) (*model.Build, error) {
+	key := metadataKeyForBuildId(id)
+	reader, err := b.Get(ctx, key)
+	if err != nil {
+		return nil, errors.Wrapf(err, "fetching build metadata for build '%s'", id)
+	}
+
+	bytes, err := ioutil.ReadAll(reader)
+	if err != nil {
+		return nil, errors.Wrapf(err, "reading build metadata for build '%s'", id)
+	}
+	metadata := buildMetadata{}
+	err = metadata.fromJSON(bytes)
+	if err != nil {
+		return nil, errors.Wrapf(err, "parsing build metadata for build '%s'", id)
+	}
```

Comment:
Maybe it makes sense to use a json [Decoder](https://pkg.go.dev/encoding/json#NewDecoder) since it takes care of reading/unmarshaling from a reader?

Reply:
Thanks for the link to the docs!